### ISSUE TITLE
Errors in console when fuzz testing debug version

### ISSUE THREAD

**Issue by Nancy-Salpepi** [phetsims/area-model-algebra#14]
**Test device**
MacBook Air M1 chip

**Operating System**
14.3.1

**Browser**
Safari 17.3.1 and Chrome

**Problem description**
For https://github.com/phetsims/qa/issues/1047 and https://github.com/phetsims/qa/issues/1048 with the debug version of the sim and ?fuzz I am seeing these errors in the console which weren't there for rc.1: 

<img width="504" alt="Screenshot 2024-02-27 at 9 13 57 AM" src="https://github.com/phetsims/area-model-algebra/assets/87318828/15cebaad-83d4-4838-b3c9-a5c3883fa446">

<img width="504" alt="Screenshot 2024-02-27 at 9 14 58 AM" src="https://github.com/phetsims/area-model-algebra/assets/87318828/e7bc278d-6aef-4acd-8e80-f4e734c45799">
<img width="504" alt="Screenshot 2024-02-27 at 9 15 24 AM" src="https://github.com/phetsims/area-model-algebra/assets/87318828/e8c3c244-74db-411d-88dc-df04a9294bf3">
<img width="504" alt="Screenshot 2024-02-27 at 9 29 44 AM" src="https://github.com/phetsims/area-model-algebra/assets/87318828/2edf8173-471f-41e9-ab6e-98705bf1824f">

<details>
Troubleshooting information:
!!!!! DO NOT EDIT !!!!!
Name: ‪Area Model Algebra‬
URL: https://phet-dev.colorado.edu/html/area-model-algebra/1.3.0-rc.2/phet/area-model-algebra_all_phet_debug.html
Version: 1.3.0-rc.2 2024-02-26 18:51:51 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.3.1 Safari/605.1.15
Language: en-US
Window: 1428x712
Pixel Ratio: 2/1
WebGL: WebGL 1.0
GLSL: WebGL GLSL ES 1.0 (1.0)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 16384x16384
OES_texture_float: true
Dependencies JSON: {}
</details>

**Comment by marlitas** [comment]
To reproduce:
1. Go to the first screen
2. Click on the "100x100" rectangular radio button.

**Comment by Luisav1** [comment]
I wasn't able to reproduce this on main but did trace the dispose error to `RichDragListener` in `ProportionalPartitionLineNode`.
I also traced the reentry detected error to `SceneRadioButtonGroup` and suspect that it might be a result of not being able to dispose the drag handler (but cannot confirm this).

@jbphet had the idea to create a build of the RC branch to reproduce it there. We were able to trace that the RC had the version of `RichDragListener` where `isDisposable: false` temporarily. I applied a patch for https://github.com/phetsims/sun/commit/55f1190c795a3962c66558e5bad3d1d0ae84d95a that fixed the dispose error to the 1.3 RC branches. So the errors shouldn't appear anymore.
